### ISSUE TITLE
Gpus were used

### ISSUE THREAD

**Issue by north151** [AlonzoLeeeooo/StableV2V#11]
How many Gpus were used in the training process?

**Comment by AlonzoLeeeooo** [comment]
Hi @north151 ,

Thanks for your attention. Would you please specify which training process do you refer to? Here are some training details of different components in `StableV2V`. Hope these information might address your concerns.

For the PFE and CIG, i.e., the first-frame image editor and image-to-video generator, we directly leverage off-the-shelf pre-trained model weights, and do not need to perform training. To download these model weights, you may refer to [our HuggingFace model repository](https://huggingface.co/AlonzoLeeeooo/StableV2V) or their official links described in [this section](https://github.com/AlonzoLeeeooo/StableV2V?tab=readme-ov-file#2-pre-trained-model-weights). Also note that the motion simulation and the depth simulation processes of ISA are learning-free algorithms, so herein we do not need to do any form of training.

Particularly, for the shape-guided depth refinement network in ISA, we use [YouTube-VOS](https://codalab.lisn.upsaclay.fr/competitions/6066#participate-get_data) as our training data. We recommend you to use at least two NVIDIA 3090 GPUs to conduct the training. Besides, please refer to the instructions of data preparation and training in [this section](https://github.com/AlonzoLeeeooo/StableV2V?tab=readme-ov-file#training-of-the-shape-guided-depth-refinement-network) of our repository.

If you have any further questions, please do not hesitate to contact us in this issue.

Best regards,
Chang

**Comment by AlonzoLeeeooo** [comment]
> Thank you. I mean the shape-guided depth refinement network

Get it! Please follow the instructions below to train the shape-guided depth refinement network:
## 1. Download the `YouTube-VOS` Dataset
We use `YouTube-VOS` to conduct the training process of our shape-guided depth refinement network.
Before you start the training process, you need to first download its source videos and annotations from [this link](https://codalab.lisn.upsaclay.fr/competitions/6066#participate-get_data).
Once downloaded, the data follows the structures below:
```
youtube-vos
├── JPEGImages                     <----- Path of source video frames
├── Annotations                    <----- Path of segmentation masks
└── meta.json                      <----- Annotation file for the segmentation masks
```

## 2. Use `MiDaS` to Annotate the Depth Maps
Once the video frames are ready, the next step is to annotate their corresponding depth maps.
Specifically, make sure you download the `MiDaS` model weights from [this link](https://github.com/isl-org/MiDaS/releases/download/v3_1/dpt_swin2_large_384.pt).
Then, you can execute the following command lines with our automatic script:
```bash
python scripts/extract_youtube_vos_depths.py --midas-path checkpoints/dpt_swin2_large_384.pt --dataset-path data/youtube-vos/JPEGImages --outdir data/youtube-vos/DepthMaps
```

## 3. Use `U2-Net` to Annotate the First-frame Shape Masks
Our depth refinement network uses an additional network channel to take the first-frame shape mask as guidance, thus you need to annotate them for the `YouTube-VOS` dataset.
First, make sure you download the `U2-Net` model weights from [this link](https://huggingface.co/AlonzoLeeeooo/LaCon/resolve/main/data-preprocessing/u2net.pth).
Then, you execute the following command lines with our automatic script:
```bash
python scripts/extract_youtube_vos_shapes.py --video-root data/youtube-vos/JPEGImages --model-dir checkpoints/u2net.pth --outdir data/youtube-vos/FirstFrameMasks
```


## 4. Train the Model
Finally, you are ready to execute the training process with the following command line:
```bash
python train_completion_net.py --video-path data/youtube-vos/JPEGImages --shape-path data/youtube-vos/FirstFrameMasks --max-train-steps 50000 --outdir results/shape-guided-depth-refinement --checkpoint-freq 2000 --validation-freq 200
```

The trained model weights will be saved at `results/checkpoints`, and the visualizations of intermediate results can be checked via `tensorboard`, with the logs saved at `results/tensorboard`.

**Comment by AlonzoLeeeooo** [comment]
> I have downloaded the data set, but I currently do not have 80G memory to complete the inference, do I want to know that the input image must be equal in length and width? Is this determined by i2vGen-XL?

No, you needn't. Theoretically you can configure it to any height and width through `--height` and `--width`, as long as the height and width can be divided by 8 (it needs to properly encode by VAE).

Besides, you can also try decreasing the generated frames by configuring `--n-sample-frames` to less than `16`. This will also lead to less memory requirement.

**Comment by AlonzoLeeeooo** [comment]
Hi @north151 ,

I personally recommend you use at least one A40 GPU (>48 GB). According to our previous experiences, training videos with  heights and widths of 384 needs two NVIDIA 3090 GPUs, so 720P and 1080P definitely needs more memory to do the training.

The reasoning (inference) process won't take much memory as the training does. I think that >12 GB GPU memory should be sufficient.
